### PR TITLE
feat: Add event markers

### LINKED ISSUE #14618: Add giraffe status/notification-event layer to visualizations.
This issue is not required for alerts mvp. 
When a user views a cell with a check- they currently see data from the query the check is running on, and the thresholds defined on the check.
We would like to add a vis layer that displays vertical lines indicating check status events.
Perhaps on hover on these lines, the legend displays information about the check that produced the status and a link to the status history page for that check? 
We should create this visualization layer that it might be used in the upcoming annotations feature.

### PR DESCRIPTION
**Pull request by ebb-tide**
Closes #14618

### CODE REVIEW

File: ui/package.json [hunk 1]
```diff
@@ -160,7 +160,7 @@
   "dependencies": {
     "@influxdata/clockface": "0.0.28",
     "@influxdata/flux-parser": "^0.3.0",
-    "@influxdata/giraffe": "0.16.3",
+    "@influxdata/giraffe": "0.16.4",
```
Comment (alexpaxton) [inline ui/package.json]: Yay

File: ui/src/alerting/constants/index.ts [hunk 2]
```diff
@@ -40,11 +40,14 @@ export const CHECK_OFFSET_OPTIONS: DurationOption[] = [
   {duration: '30d', displayText: '30 days'},
 ]
 
+export const MONITORING_BUCKET = '_monitoring'
```
Comment (alexpaxton) [inline ui/src/alerting/constants/index.ts]: Yay

File: ui/src/shared/components/EventMarkers.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
+// Libraries
+import React, {FunctionComponent} from 'react'
+
+// Components
+import EventMarker from 'src/shared/components/EventMarker'
+
+// Utils
```
Comment (alexpaxton) [inline ui/src/shared/components/EventMarkers.tsx]: Probs don't need this comment

File: ui/src/alerting/utils/statusEvents.ts [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  orgID: string,
+  checkID: string,
+  extern: File
+): CancelBox<StatusRow[][]> => {
+  const query = `
+from(bucket: "${MONITORING_BUCKET}")
+  |> range(start: v.timeRangeStart, stop: v.timeRangeStop)
+  |> filter(fn: (r) => r._measurement == "statuses" and r._field == "_message")
+  |> keep(columns: ["_time", "_value", "_check_id", "_check_name", "_level"])
+  |> rename(columns: {"_time": "time",
```
Comment (desa) [inline ui/src/alerting/utils/statusEvents.ts]: *nit* If we put the rename at the very end of things, I think it cleans things up a tiny bit.


```
from(bucket: "${MONITORING_BUCKET}")
  |> range(start: v.timeRangeStart, stop: v.timeRangeStop)
  |> filter(fn: (r) => r._measurement == "statuses" and r._field == "_message")
  |> keep(columns: ["_time", "_value", "_check_id", "_check_name", "_level"])
  |> window(every: 1s)
  |> group(columns: ["_start", "_stop"])
  |> filter(fn: (r) => r.check_id == "${checkID}")
  |> rename(columns: {"_time": "time",
                      "_value": "message",
                      "_check_id": "checkID",
                      "_check_name": "checkName",
                      "_level": "level"})
```

**Pull request merged**